### ISSUE TITLE
dask worker is not saving in correct path

### ISSUE THREAD

**Issue by RodrigoNeves95** [dask/dask#5210]
# Versions 
dask==2.2.0
distributed==2.2.0

## code
```python
import dask
import dask.dataframe as dd

import pandas as pd
import numpy as np

from dask.distributed import Client
client = Client('localhost:32000')
client

import os
import datetime

df = dask.datasets.timeseries(freq='1s')
df

if not os.path.exists('data2'):
    os.mkdir('data2')

def name(i):
    """ Provide date for filename given index

    Examples
    --------
    >>> name(0)
    '2000-01-01'
    >>> name(10)
    '2000-01-11'
    """
    return str(datetime.date(2000, 1, 1) + i * datetime.timedelta(days=1))

df.to_csv('data2/*.csv', name_function=name)
df = dd.read_csv('data2/2000-*-*.csv')
```

This code does not work in dask==2.2.0 but works in dask==2.1.0 (both with distributed 2.2.0, and that's why I'm opening the issue here). I should emphasize that I'm running this in a distributed cluster (kubernetes) but the same thing would happen would probably happen if I would run a LocalCluster where the workers would be initialized in a different path where the scheduler is running.

Basically what's happening is that each worker is creating a `data2/*.csv` on his "home" path and not in the path referenced by the scheduler (as it was done in dask==2.1.0)

**Comment by martindurant** [comment]
You would be better off using absolute paths every time to avoid the ambiguity. What path is set as the process CWD depends on the method to launch the process.

**Comment by mrocklin** [comment]
I don't think that we can expect users to know this ahead of time.
Historically we've converted things to absolute paths for them.  Is this no
longer the case?

**Comment by martindurant** [comment]
The FS implementation does convert to absolute path for all operations ([code](https://github.com/intake/filesystem_spec/blob/master/fsspec/implementations/local.py#L141)), so actually I can't see how the path remains relative. Could it be
```
if not os.path.exists('data2'):
    os.mkdir('data2')
```
the *directory* is being made in different places and then not found? This is circumventing anything that Dask or fsspec does.

**Comment by mrocklin** [comment]
The directory is being made only once, here in this process.

Then we call `to_csv` with that same relative path.  We should turn it into an absolute path at this point and make sure that the paths in the task graph are absolute.  Same with `read_csv`.

**Comment by martindurant** [comment]
Also true for a kubernetes cluster?

Trying:
```
In [11]: df = dd.read_csv("2006*.csv", blocksize=None)
In [13]: df.dask['read-block-0-52bb4da1715feca8b6eb7e995d19ed9f']
Out[13]:
(<function dask.bytes.core.read_block_from_file(lazy_file, off, bs, delimiter)>,
 <OpenFile '/Users/mdurant/data/airlines-csv/2006.csv'>,
 0,
 None,
 b'\n')

In [14]: com = df.to_csv('out*.csv', compute=False)
In [18]: com[0].dask['_write_csv-49902057-c47d-4b86-a5cc-8b4bf2bdd1f2']
Out[18]:
(<function dask.dataframe.io.csv._write_csv(df, fil, **kwargs)>,
 ('from-delayed-513aaa950376a34ebd5df7171ae76f0f', 0),
 <OpenFile '/Users/mdurant/data/airlines-csv/out0.csv'>)
```
the paths are all absolute.

**Comment by mrocklin** [comment]
Well then I don't know.  @RodrigoNeves95 are your machines on the same filesystem?

**Comment by RodrigoNeves95** [comment]
Yes, I'm currently using CephFS. What for me is most difficult to understand is why this is currently working for me in `dask==2.1.0` but not in `dask==2.2.0`. I have been using relative paths for a long time and everything should work. And I agree with @mrocklin that user should not know that in advance. 

Something that I did not check was FS versions. I will do it and give more info here!

**Comment by RodrigoNeves95** [comment]
I'm leaving here dask graph output

## dask 2.2.0 + fsspec 0.4.0
```python
save_df = df.to_csv('data2/*.csv', name_function=name, compute=False)

save_df[0].dask['_write_csv-ee04281c-41c1-4824-a2c6-de531ef0de94']

(<function dask.dataframe.io.csv._write_csv(df, fil, **kwargs)>,
 ('make-timeseries-808a2fd41a385a1db4e0cc435b7ba44e', 0),
 <OpenFile 'data2/2000-01-01.csv'>)
```

## dask 2.1.0
```python
save_df = df.to_csv('data2/*.csv', name_function=name, compute=False)

save_df[0].dask['_write_csv-0bb3d92a-ff23-425d-aeca-bcc4a44f0d97']
(<function dask.dataframe.io.csv._write_csv(df, fil, **kwargs)>,
 ('make-timeseries-f9b7bf1ff69a38252e07099d4337b840', 0),
 <dask.bytes.core.OpenFile at 0x7fd4202c5110>)
```

It's possible to see that dask 2.2.0 is using a relative path (different behaviour from @martindurant).  Which other libs can have an impact on this?

**Comment by martindurant** [comment]
OK, I have a reproducer and I'm looking at it now.
